src/main/java/ma/hibernate/dao/PhoneDaoImpl.java

```diff
@@ -12,11 +19,49 @@ public PhoneDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Phone create(Phone phone) {
-        return null;
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(phone);
+            transaction.commit();
+            return phone;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't create phone " + phone, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Phone> findAll(Map<String, String[]> params) {
-        return null;
+        try (Session session = factory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Phone> query = criteriaBuilder.createQuery(Phone.class);
+            Root<Phone> root = query.from(Phone.class);
+            List<Predicate> andPredicates = new ArrayList<>();
+            for (Map.Entry<String, String[]> entry: params.entrySet()) {
+                String key = entry.getKey();
+                String[] value = entry.getValue();
+                List<Predicate> orPredicates = new ArrayList<>();
+                for (int i = 0; i < value.length; i++) {
+                    Predicate orPredicate = criteriaBuilder.equal(root.get(key), value[i]);
+                    orPredicates.add(orPredicate);
+                }
+                andPredicates.add(criteriaBuilder.or(orPredicates.toArray(new Predicate[]{})));
+            }
+            Predicate finalPredicate =
+                    criteriaBuilder.and(andPredicates.toArray(new Predicate[]{}));
+            CriteriaQuery<Phone> criteriaQuery = query.select(root).where(finalPredicate);
+            return session.createQuery(criteriaQuery).getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't get all Pnones from DB", e);
```

Comment:
Maybe add params to error to know what parameters caused failure

Reply:
fix